app/database/dataStore.go

```diff
@@ -49,8 +51,23 @@ func (s *DataStore) GroupAncestors() *GroupAncestorStore {
 	return &GroupAncestorStore{s}
 }
 
-func generateID() int64 {
+// NewID generates a positive random int64 to be used as ID
+// !!! To be safe, the insertion should be be retried if the ID conflicts with an existing entry
+func (s *DataStore) NewID() int64 {
 	// gen a 63-bits number as we want unsigned number stored in a 64-bits signed DB attribute
-	// !!! to be safe, the insertion should be be retried if the ID conflicts with an existing entry
 	return rand.Int63()
 }
+
+// GenIDIfNotSet does check the given ID is set. If not, generate a (random) ID for it
+func (s *DataStore) GenIDIfNotSet(id *t.Int64) {
+	if !id.Set {
+		*id = *t.NewInt64(s.NewID())
+	}
+}
+
+// InTransaction execute the given function in a transaction and commit
+func (s *DataStore) InTransaction(txFunc func(*DataStore) error) error {
+	return s.db.inTransaction(func(db *DB) error {
+		return txFunc(&DataStore{db})
```

Comment:
Use names when assigning values to struct fields.

Reply:
on external package it is important (required?). Here, I don't see how `&DataStore{db:db}` makes it more readable than `&DataStore{db}`

Reply:
There is a `go vet` warning: https://golang.org/cmd/vet/#hdr-Unkeyed_composite_literals

Reply:
How do you get this warning? I do not have it locally, also it is not raised by golangci (which includes go vet) and `make lint` which includes go vet. Are you on Go 1.11 ?

Reply:
Hm, I used to get such warnings i VSCode and still do for primitive values, but here is doesn't show up indeed. Never mind then.